opal/corelib/range.rb

```diff
@@ -85,16 +93,26 @@ def exclude_end?
     @exclude
   end
 
-  alias :first :begin
+  def first(n=(n_specified=true; nil))
```

Comment:
`== null` would be enough here.

`undefined` is a special value of any optional argument (https://github.com/opal/opal/blob/master/lib/opal/nodes/args/optarg.rb#L17). It compiles to nothing, so your argument is `undefined` when you pass no value to your method.

``` js
function compiledOpalMethod(optional_argument) {
  console.log(optional_argument);
}

compiledOpalMethod();
// => undefined
```